### ISSUE TITLE
Nondescriptive error when republishing after changing dependency versions

### ISSUE THREAD

**Issue by Qqwy** [hexpm/hex#392]
I just published an updated version of my library, and then realized that I had not updated the version number of one of the dependencies in Mix.exs.

After changing it and then attempting to re-publish (republishing is possible for up to an hour after the original publishment), Hex gives me the following stack trace:

```elixir
Docs successfully generated.
View them at "doc/index.html".
Publishing package...
[#########################] 100%
Publishing failed
Validation error(s)
  requirements:
** (ArgumentError) argument error
    :erlang.bit_size(nil)
    (hex) lib/hex/utils.ex:137: anonymous fn/2 in Hex.Utils.pretty_errors/2
    (elixir) lib/enum.ex:682: anonymous fn/3 in Enum.each/2
    (stdlib) lists.erl:1263: :lists.foldl/3
    (elixir) lib/enum.ex:1829: Enum.each/2
    (elixir) lib/enum.ex:682: anonymous fn/3 in Enum.each/2
    (stdlib) lists.erl:1263: :lists.foldl/3
    (elixir) lib/enum.ex:1829: Enum.each/2
```

Now, the fact that you're not allowed to republish something when the dependency versions changed is not strange, but I think Hex is supposed to show a slightly more descriptive error message here.

**Comment by wojtekmach** [comment]
@Qqwy looking at the error:

```
Validation error(s)
  requirements:
** (ArgumentError) argument error
    :erlang.bit_size(nil)
```

it looks like it's related to `requirements`. When the 1h window is up we send the error in the `inserted_at` key. Could you send github url for your package and/or paste `deps` part of your mix.exs that you were trying to publish? (support@hex.pm if it's private.) That would be a huge help to reproduce and fix this bug, we definitely shouldn't crash.

**Comment by Qqwy** [comment]
@wojtekmach thank you for your reply!

I encountered the issue when releasing one of the prerelease versions of  [Prioqueue](https://github.com/Qqwy/elixir-prioqueue), but I was unable to quickly find the exact commits at which it happened.

I think it is possible to reproduce the problem as follows:
_(Note that this does involve publishing a package to Hex; but you probably have a way to fake that.)_

---

```
$ mix new foo
$ cd foo
```
Now edit `mix.exs` to contain:

```
  def project do
    [
      app: :foo,
      version: "0.1.0",
      elixir: "~> 1.3",
      start_permanent: Mix.env == :prod,
      deps: deps(),
      name: "Foo",
      description: description(),
      package: package()
    ]
  end

defp deps()
  [
    {:extractable, "~> 0.1.0"},
    {:insertable, "~> 0.1.0"}
  ]

  defp description do
    """
    Hex Testing package, issue #392.
    """
  end

  defp package() do
    [
      name: :foo,
      files: ["lib", "mix.exs", "README*"],
      maintainers: ["Qqwy"],
      licenses: ["Apache 2.0"],
      links: %{}
    ]
  end
end
```

And now run `mix hex.publish`.

After this, alter the `deps` in `mix.exs` to:

```
defp deps()
  [
    {:extractable, "~> 0.2.0"},
    {:insertable, "~> 0.2.0"}
  ]
```

And now run `mix hex.publish` again, which should attempt to republish the same version and cause the error because the dependencies have changed.

**Comment by wojtekmach** [comment]
Thank you @Qqwy, I was able to reproduce this. Indeed the problem happens when a package was released with dependency say `foo ~> 0.1.0` and then we re-publish with `foo ~> 0.2.0`
